packages/inspection-capture-web/src/PhotoCapture/hooks/useComplianceAnalytics.ts

```diff
@@ -0,0 +1,55 @@
+import { useAnalytics } from '@monkvision/analytics';
+import { useMonkState } from '@monkvision/common';
+import { Image, ImageStatus, Sight } from '@monkvision/types';
+import { useEffect, useState } from 'react';
+
+interface ImageEventTracking extends Image {
+  isAlreadySent: boolean;
+}
+
+/**
+ * Parameters of the useComplianceAnalytics hook.
+ */
+export interface ComplianceAnalyticsParams {
+  /**
+   * The inspection ID.
+   */
+  inspectionId: string;
+  /**
+   * The list of sights passed to the PhotoCapture component.
+   */
+  sights: Sight[];
+}
+
+/**
+ * Custom hook used for the compliance analytics by sending an event for 'non_compliant' image.
+ */
+export function useComplianceAnalytics({ inspectionId, sights }: ComplianceAnalyticsParams) {
+  const [imagesEventTracking, setImagesEventTracking] = useState<ImageEventTracking[]>([]);
+  const { state } = useMonkState();
+  const { trackEvent } = useAnalytics();
+
+  useEffect(() => {
+    const newImagesEventTracking = state.images
+      .filter((image) => image.inspectionId === inspectionId && image.sightId)
+      .map((image) => {
+        const imageEventTracking = imagesEventTracking.find((i) => i.id === image.id);
+        if (imageEventTracking?.isAlreadySent) {
+          return imageEventTracking;
+        }
+        if (image.status === ImageStatus.NOT_COMPLIANT && image.complianceIssues) {
+          trackEvent('Compliance Issue', {
+            ...Object.fromEntries(
+              Object.values(image.complianceIssues).map((issue) => [issue, true]),
+            ),
+            sightId: image.sightId,
+            sightLabel: sights.find((sight) => sight.id === image.sightId)?.label,
+          });
```

Comment:
I think we should only keep track of the compliance issue that is displayed to the user AKA the first one in the array.

Reply:
Ok so now we'll have a tag with: `complianceIssue: 'Issue name'`